### PR TITLE
Almost complete admin panel

### PR DESCRIPTION
**Pull request by margawron**
Still waiting for #60 to merge, can merge this now, will implement rest later.

### CODE REVIEW

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java [hunk 1]
```diff
@@ -46,10 +45,10 @@ public ImageController(final ImageService imageService, final PermissionEvaluato
     @PreAuthorize("@permissionEvaluatorService.hasPrivilege(authentication, 'image', 'post' )")
     public ResponseEntity postImage(@Image(service = ImageService.class) final MultipartFile file) throws IOException {
         final String entityCode = imageService.save(new ImageModelDto(null, file.getOriginalFilename(), file.getBytes()));
-        final HttpHeaders headers = new HttpHeaders();
+        final HashMap<String,String> response = new HashMap<>(1);
         final URI uri = linkTo(methodOn(this.getClass()).getSingle(entityCode)).toUri();
-        headers.add("Location", uri.toASCIIString());
-        return new ResponseEntity(headers, HttpStatus.CREATED);
+        response.put("self", uri.toASCIIString());
```
Comment (syphzur) [inline PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java]: due to RFC 2616 (current definiton of Hypertext Transfer Protocol standard) POST response should have empty body and Location header

**Review by syphzur — CHANGES_REQUESTED** [review]

File: PhotoPlusFrontend/PhotoPlus/src/app/components/admin-panel/subcomponents/change-privileges/change-privileges.component.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export class ChangePrivilegesComponent implements OnInit {
     this.filteredUsers = new Array<BehaviorSubject<User>>();
     this.userService.getAll().subscribe((data) => {
       for (let user of data) {
-        if(user.name === "admin"){
+        if(user.login === "admin"){
```
Comment (syphzur) [inline PhotoPlusFrontend/PhotoPlus/src/app/components/admin-panel/subcomponents/change-privileges/change-privileges.component.ts:27]: maybe (user.role=== "admin")?

File: PhotoPlusFrontend/PhotoPlus/src/app/services/image/image.service.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+export class ImageService {
+
+  protected hostAddress = environment.hostAddress;
+
+  constructor(private httpClient: HttpClient) { }
+
+  post(file: File){
+    let formData = new FormData();
+    formData.append('file',file,file.name);
+    return this.httpClient.request<ImageResponse>('POST',this.hostAddress + "image",{body: formData});
```
Comment (syphzur) [inline PhotoPlusFrontend/PhotoPlus/src/app/services/image/image.service.ts]: you can easily convert that to standard post request with location header

Comment (syphzur) [reply PhotoPlusFrontend/PhotoPlus/src/app/services/image/image.service.ts]: with using abstract service

File: PhotoPlusFrontend/PhotoPlus/src/app/components/admin-panel/subcomponents/change-privileges/change-privileges.component.ts (hunk 2 above)
Comment (margawron) [reply PhotoPlusFrontend/PhotoPlus/src/app/components/admin-panel/subcomponents/change-privileges/change-privileges.component.ts:27]: No, i wanted to disallow changing the role of builtin user.

File: PhotoPlusFrontend/PhotoPlus/src/app/services/image/image.service.ts (hunk 3 above)
Comment (margawron) [reply PhotoPlusFrontend/PhotoPlus/src/app/services/image/image.service.ts]: File posting does not work with httpClient correctly.
Edit: Or at least what i would call corretly.

File: PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java (hunk 1 above)
Comment (margawron) [reply PhotoPlusBackend/photoplus/src/main/java/pl/polsl/photoplus/controllers/api/ImageController.java]: Fixed in a487057 (backend) and in 4813b1b (frontend).

**Review by syphzur — APPROVED** [review]

**Pull request merged**